files/chefdk-envhacks/windows/chefdk_envhacks.rb

```diff
@@ -0,0 +1,49 @@
+## Environment hacks for running Ruby with ChefDK ##
+# ENV['HOME'] is not set by default on Windows. We need to set this to
+# something sensible since a lot of Ruby code depends on it. It is important 
+# for this directory to exist and be available, so we are introducing logic
+# here to pick a working HOME
+#
+# You can find this file the repo at https://github.com/chef/omnibus-chef
+
+if !ENV['HOME'] || !File.exists?(ENV['HOME'])
+  old_home = ENV['HOME']
+  found = false
+  alternate_homes = ["#{ENV['HOMEDRIVE']}#{ENV['HOMEPATH']}", ENV['USERPROFILE']]
+
+  alternate_homes.each do |path|
+    if File.exists?(path)
+      ENV['HOME'] = path
+      found = true
+      break
+    end
+  end
+
+  STDERR.puts <<-EOF
+The HOME environment variable (#{old_home}) was not set to a valid value.
```

Comment:
Mostly a language issue here: the fact that the parenthetical phrase is preceded by the word "variable" implies that the word in parentheses refers to the variable, not the value of that variable, IMO. It would be more clear to me if the phrase were something like "The HOME environment variable is set to an invalid value (#{old_home})" or some language where the use of variable vs. value is unambiguous.